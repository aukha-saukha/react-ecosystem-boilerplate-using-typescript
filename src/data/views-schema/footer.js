const FOOTER_LINKS = [
  {
    dataTest: 'about-us',
    linkToLocation: '/about-us',
    name: 'About us',
  },
  {
    dataTest: 'contact-us',
    linkToLocation: '/contact-us',
    name: 'Contact us',
  },
];

module.exports = {
  FOOTER_LINKS,
};
