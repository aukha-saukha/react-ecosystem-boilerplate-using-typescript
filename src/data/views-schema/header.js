const LOGO = {
  linkToLocation: '/',
  name: 'REB',
};

module.exports = {
  LOGO,
};
