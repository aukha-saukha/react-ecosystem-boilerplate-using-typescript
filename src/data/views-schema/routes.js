const ROUTES = [
  {
    componentName: 'AboutUs',
    exact: true,
    pathToMatch: '/about-us',
  },
  {
    componentName: 'ContactUs',
    exact: true,
    pathToMatch: '/contact-us',
  },
  {
    componentName: 'Home',
    exact: true,
    pathToMatch: '/',
  },
];

module.exports = {
  ROUTES,
};
