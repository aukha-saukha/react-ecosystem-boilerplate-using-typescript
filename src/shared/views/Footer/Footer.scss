@import '../../../static/css/_variables';

.footer {
  border-top: 1px solid map-get($border-color, footer-top);
  grid-area: footer;
}

.footer li:not(:last-child) {
  margin-right: 15px;
}

.footer a {
  color: map-get($text-color, link-footer);
  font-size: 1.6rem;
}
