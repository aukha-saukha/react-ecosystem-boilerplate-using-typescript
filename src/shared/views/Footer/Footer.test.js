import { mount, shallow } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { FOOTER_LINKS } from '../../../data/views-schema/footer';
import Footer from './FooterComponent';

describe('Footer component', () => {
  it('has all navigation links defined in the footer schema', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    // Check if number of footer links is equivalent of links defined in footer schema
    expect(wrapper.find('a')).toHaveLength(FOOTER_LINKS.length);

    // Grab all link using their href property, and confirm their text matches to their respective
    // values in footer schema.
    FOOTER_LINKS.forEach((footerLink) => {
      expect(wrapper.find({ href: footerLink.linkToLocation }).text()).toEqual(
        `${footerLink.name}`
      );
    });

    wrapper.unmount();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Footer />);

    expect(wrapper).toMatchSnapshot();
  });
});
