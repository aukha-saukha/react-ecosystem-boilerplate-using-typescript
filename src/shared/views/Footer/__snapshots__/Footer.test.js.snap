// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  className="display-flex flex-items-align-center padding-all-15 footer"
>
  <nav>
    <ul
      className="display-flex"
    >
      <li
        key="About us"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          data-test="about-us"
          to="/about-us"
        >
          About us
        </NavLink>
      </li>
      <li
        key="Contact us"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          data-test="contact-us"
          to="/contact-us"
        >
          Contact us
        </NavLink>
      </li>
    </ul>
  </nav>
</footer>
`;
